sandbox.go

```diff
@@ -644,13 +644,6 @@ func (sb *sandbox) SetKey(basePath string) error {
 	sb.Lock()
 	sb.osSbox = osSbox
 	sb.Unlock()
-	defer func() {
-		if err != nil {
-			sb.Lock()
```

Comment:
> Its generally a good practice to cleanup any locally allocated resource 

Agree, but the netns is not a locally allocated resource.

SetKey job is in fact to informing libnetwork about the existing netns for the specific Sandbox. So libnetwork must first link the two and this must not change. This is what GetSandboxForExternalKey does, creating the lbnetwork abstraction, handle for the existing netns.

If we instead cleanup what GetSandboxForExternalKey did, then we will loose the handle and we won't be able to remove the netns when docker invokes the Sandbox.Delete().